### DISCUSSION TITLE
Multi-tasking: combining survey text and keyboard response plugins

### DISCUSSION THREAD

**Discussion by mbeasle2** [jspsych/jsPsych#3170] [Q&A]
Hello, I have two different tasks that I need to combine to be a single task (it's a multi-tasking type experiment). For the first task, I used the survey text plugin to make a single trial a random math question. The task deducts part of a point for the amount of time that it takes the participant to answer; if they are quick enough they can earn 1 point per question. The question changes after it is answered or after 15 seconds. 

The second task is a RT task using the image keyboard response plugin. A square appears on the screen. I have it stay there for a set amount of time (4 min) and about every 30s when the stimulus is present, the participant must hit the space bar. They can earn a point for each time this is done. Eventually the stimulus will go away. It will come back in a about a minute or two, at random. 

I want these to happen independently of one another at the same time. It was suggested to make the first task then edit the plugin to allow for the second task. I am a little confused on how to get started doing this. My first instinct is to copy the information from the second plugin into the first one, to allow for all the parameters I will need in the second plugin, as well. But I run into problems when I am doing it this way. Any suggestions?

**Comment by ChristopheBossens** [comment]
Hi,

This seems to be a challenging problem. My first instinct was to create two jsPsych instances, each rendered into a separate display_element, but it feels that jsPsych is not really meant to do this (a lot of components have a unique id, which would no longer be the case if you have two instances).

Maybe you can try to make it work by having each experiment in a separate html file and using two i-frames to run the simultaneously?

Otherwise I don't really see how to solve it, as it basically means running two timelines in parallel? Is there a way of defining the concept of a single trial at the level of the whole experiment? From what I understand a single trial is a 4 minute presentation of a square in which every 30s you present a square which requires a keyboard response, intermixed with math questions?
